### PR TITLE
CMakeDeps imported target multiconfig

### PR DESCRIPTION
**Pull request by lasote**
Changelog: omit
Docs: omit

### CODE REVIEW

File: conan/tools/cmake/cmakedeps/templates/macros.py [hunk 1]
```diff
@@ -41,7 +41,7 @@ def template(self):
            endif()
        endmacro()
 
-       function(conan_package_library_targets libraries package_libdir deps out_libraries out_libraries_target config_suffix package_name)
```
Comment (lasote) [inline conan/tools/cmake/cmakedeps/templates/macros.py:44]: `out_libraries` arg was not used.

File: conan/tools/cmake/cmakedeps/templates/target_configuration.py [hunk 2]
```diff
@@ -60,64 +61,43 @@ def template(self):
...
         # We need to add our requirements too
-        set({{ pkg_name }}_LIBRARIES_TARGETS{{ config_suffix }} {{ '"${' }}{{ pkg_name }}_LIBRARIES_TARGETS{{ config_suffix }}{{ '};' }}{{ deps_targets_names }}")
-        set({{ pkg_name }}_LIBRARIES{{ config_suffix }} {{ '"${' }}{{ pkg_name }}_LIBRARIES{{ config_suffix }}{{ '};' }}{{ deps_targets_names }}")
+        list(APPEND {{ pkg_name }}_LIBRARIES{{ config_suffix }} {{ deps_targets_names }})
 
         # FIXME: What is the result of this for multi-config? All configs adding themselves to path?
         set(CMAKE_MODULE_PATH {{ '${' }}{{ pkg_name }}_BUILD_DIRS{{ config_suffix }}} {{ '${' }}CMAKE_MODULE_PATH})
         set(CMAKE_PREFIX_PATH {{ '${' }}{{ pkg_name }}_BUILD_DIRS{{ config_suffix }}} {{ '${' }}CMAKE_PREFIX_PATH})
 
-        {%- for comp_variable_name, comp_target_name in components_names %}
```
Comment (lasote) [inline conan/tools/cmake/cmakedeps/templates/target_configuration.py:93]: moved below, with the other components code to be easier.

**Review by memsharded — APPROVED** [review]

**Review by jcar87 — APPROVED** [review]

File: conan/tools/cmake/cmakedeps/templates/target_configuration.py [hunk 3]
```diff
@@ -60,64 +61,44 @@ def template(self):
...
 
         foreach(_SYSTEM_LIB {{ '${' }}{{ pkg_name }}_SYSTEM_LIBS{{ config_suffix }}})
-            list(APPEND {{ pkg_name }}_LIBRARIES_TARGETS{{ config_suffix }} ${_SYSTEM_LIB})
             list(APPEND {{ pkg_name }}_LIBRARIES{{ config_suffix }} ${_SYSTEM_LIB})
         endforeach()
 
         # We need to add our requirements too
-        set({{ pkg_name }}_LIBRARIES_TARGETS{{ config_suffix }} {{ '"${' }}{{ pkg_name }}_LIBRARIES_TARGETS{{ config_suffix }}{{ '};' }}{{ deps_targets_names }}")
-        set({{ pkg_name }}_LIBRARIES{{ config_suffix }} {{ '"${' }}{{ pkg_name }}_LIBRARIES{{ config_suffix }}{{ '};' }}{{ deps_targets_names }}")
+        set({{ pkg_name }}_LIBRARIES{{ config_suffix }} "")
```
Comment (alex-700) [inline conan/tools/cmake/cmakedeps/templates/target_configuration.py:85]: Hey, @lasote !

I think that you have broken everything here :) The variable was not cleared here before. It seems strange even when looking to several `"list(APPEND ...)"` above, which actively push something to this variable. 

I didn't understand whole picture though. But the specific problem is: `conan=1.51` broke a build of `zstd` dependant, where I'm using `${zstd_LIBRARIES}`, which become empty after update. And I'm pretty sure that bug in this PR.

Comment (lasote) [reply conan/tools/cmake/cmakedeps/templates/target_configuration.py:85]: Checking.

**Pull request merged**